### PR TITLE
chore: invert Boolean algebra

### PR DESCRIPTION
**Pull request by mlutze**
Needed since the naive translation put "Empty" as "True", which is confusing

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/phase/unification/BoolUnification.scala [hunk 1]
```diff
@@ -78,26 +78,26 @@ object BoolUnification {
             return Ok((Substitution.empty, Nil)) // 1000 hits
           }
 
-        case (Type.Cst(TypeConstructor.Empty, _), Type.Cst(TypeConstructor.Empty, _)) =>
+        case (Type.Cst(TypeConstructor.All, _), Type.Cst(TypeConstructor.Empty, _)) =>
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/unification/BoolUnification.scala:81]: Ugh. You might have to redo this performance profiling? :-/

**Comment by magnus-madsen** [comment]
Blocked on the performance specific aspects.

**Comment by magnus-madsen** [comment]
I would potentially also wait with this refactor until Bool has been fully enabled and works for relational null (I think those are strong test cases).

**Comment by mlutze** [comment]
> I would potentially also wait with this refactor until Bool has been fully enabled and works for relational null (I think those are strong test cases).

Note that relational null is currently working, just pretending to be a set for now.

**Comment by magnus-madsen** [comment]
(We can continue at a later time.)

**Pull request closed without merging**